app/controllers/concerns/defence_request_authorization.rb

```diff
@@ -1,19 +1,24 @@
 module DefenceRequestAuthorization
 
+  private
+
   def find_defence_request
+    defence_request_id = params[:defence_request_id] || params[:id]
```

Comment:
Seems sensible to let controllers decided on `id`. I experimented with a method in each controller to do that. Could go back to that approach.